### ISSUE TITLE
Enhancement: Move SonarQube step after regression tests are executed

### ISSUE THREAD

**Issue by deors** [Accenture/adop-cartridge-java#11]
Moving the SonarQube step after regression tests are executed will allow us to pull into SonarQube dashboard the information from regression test execution and from performance test execution.
On a side note, regression test execution should be done with Failsafe (plugin which is actually a copy of Surefire) bind to integration-test phase, to ensure code coverage from regression tests is reported properly in SonarQube dashboard.

**Comment by kramos** [comment]
The downside of this is that sonarqube will run much later in the pipeline i.e. after the deployment and the regression tests.

**Comment by deors** [comment]
Yes, that's correct, however it brings a lot of benefit, including the ability to have integration test coverage and overall test coverage into the quality gates (which is a very good feature).
Another option would be to do a couple of analysis:
1) A dry-run analysis after unit tests, with the outputs we have as of now in the cartridge.
2) A second, full coverage, analysis, persisted into the database, at the end of the line just before pushing to prod, as the ultimate quality gate.
Thoughts?

**Comment by nickdgriffin** [comment]
Would a dry-run analysis still enable the build to be failed if the unit test coverage or number of critical/blockers were above a threshold? Because that's the part I'd want to know about before it makes it any further as knowing I've got good integration test coverage doesn't really help me if I still have to go back and make changes to satisfy the earlier conditions.

**Comment by deors** [comment]
Dry run would work and use quality gate to break the build, at least in previous versions it worked that way. You would need to set a different quality gate i.e. to not look at IT coverage at this stage. @restalion and @viarellano were looking into this already, to ensure it works fine before sending the PR.
